app/iot/service.py

```diff
@@ -18,32 +18,32 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
         self.devices[device_id] = device
         return device_id
 
-    def unregister_device(self, device_id: str) -> None:
+    async def unregister_device(self, device_id: str) -> None:
         self.devices[device_id].disconnect()
         del self.devices[device_id]
 
-    def get_device(self, device_id: str) -> Device:
+    async def get_device(self, device_id: str) -> Device:
         return self.devices[device_id]
 
-    def run_program(self, program: list[Message]) -> None:
+    async def run_program(self, program: list[Message]) -> None:
         print("=====RUNNING PROGRAM======")
         for msg in program:
-            self.send_msg(msg)
+           await self.send_msg(msg)
         print("=====END OF PROGRAM======")
 
-    def send_msg(self, msg: Message) -> None:
-        self.devices[msg.device_id].send_message(msg.msg_type, msg.data)
+    async def send_msg(self, msg: Message) -> None:
+       await self.devices[msg.device_id].send_message(msg.msg_type, msg.data)
```

Comment:
Again, you are directly accessing the device from the dictionary and calling the send_message method on it. But what if the device_id is not present in the dictionary? This will raise a KeyError. You should handle this exception.